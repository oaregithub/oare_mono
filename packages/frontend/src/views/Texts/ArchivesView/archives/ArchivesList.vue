<template>
  <div>
    <div class="pb-1" v-for="(archive, idx) in archives" :key="idx">
      <archive-info-component :archive="archive" />
    </div>
  </div>
</template>

<script lang="ts">
import { defineComponent, PropType } from '@vue/composition-api';
import { ArchiveInfo } from '@oare/types';
import ArchiveInfoComponent from './ArchiveInfo.vue';

export default defineComponent({
  name: 'ArchivesList',
  components: {
    ArchiveInfoComponent,
  },
  props: {
    archives: {
      type: Array as PropType<ArchiveInfo[]>,
      required: true,
    },
  },
});
</script>
