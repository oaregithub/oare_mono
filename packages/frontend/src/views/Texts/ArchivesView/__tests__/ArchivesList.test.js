import Vuetify from 'vuetify';
import VueCompositionApi from '@vue/composition-api';
import { createLocalVue } from '@vue/test-utils';
import { render } from '@testing-library/vue';
import ArchivesList from '../archives/ArchivesList.vue';

const vuetify = new Vuetify();
const localVue = createLocalVue();
localVue.use(VueCompositionApi);

describe('ArchivesList test', () => {
  const createWrapper = () =>
    render(ArchivesList, {
      localVue,
      vuetify,
      stubs: ['router-link'],
      propsData: {
        archives: [
          {
            uuid: '1',
            name: 'Archive1',
            totalTexts: 6,
            totalDossiers: 2,
          },
          {
            uuid: '2',
            name: 'Archive2',
            totalTexts: 3,
            totalDossiers: 1,
          },
          {
            uuid: '3',
            name: 'Archive3',
            totalTexts: 30,
            totalDossiers: 7,
          },
          {
            uuid: '4',
            name: 'Archive4',
            totalTexts: 20,
            totalDossiers: 4,
          },
          {
            uuid: '5',
            name: 'Archive5',
            totalTexts: 15,
            totalDossiers: 8,
          },
          {
            uuid: '6',
            name: 'Archive6',
            totalTexts: 21,
            totalDossiers: 9,
          },
        ],
      },
    });

  it('matches snapshot', () => {
    expect(createWrapper()).toMatchSnapshot();
  });

  it('displays archives', () => {
    const { getByText } = createWrapper();
    [
      'Archive1 (6 texts | 2 dossiers)',
      'Archive2 (3 texts | 1 dossiers)',
      'Archive3 (30 texts | 7 dossiers)',
      'Archive4 (20 texts | 4 dossiers)',
      'Archive5 (15 texts | 8 dossiers)',
      'Archive6 (21 texts | 9 dossiers)',
    ].forEach(name => {
      expect(getByText(name));
    });
  });
});
