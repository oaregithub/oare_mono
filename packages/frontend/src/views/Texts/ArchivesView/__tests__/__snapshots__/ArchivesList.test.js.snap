// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArchivesList test matches snapshot 1`] = `
Object {
  "baseElement": <body>
    <div
      data-app="true"
    />
    <div>
      <div>
        <div>
          <router-link-stub
            to="/archives/1"
          >
            Archive1 (6 texts |
      2 dossiers)
          </router-link-stub>
        </div>
        <div>
          <router-link-stub
            to="/archives/2"
          >
            Archive2 (3 texts |
      1 dossiers)
          </router-link-stub>
        </div>
        <div>
          <router-link-stub
            to="/archives/3"
          >
            Archive3 (30 texts |
      7 dossiers)
          </router-link-stub>
        </div>
        <div>
          <router-link-stub
            to="/archives/4"
          >
            Archive4 (20 texts |
      4 dossiers)
          </router-link-stub>
        </div>
        <div>
          <router-link-stub
            to="/archives/5"
          >
            Archive5 (15 texts |
      8 dossiers)
          </router-link-stub>
        </div>
        <div>
          <router-link-stub
            to="/archives/6"
          >
            Archive6 (21 texts |
      9 dossiers)
          </router-link-stub>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div>
        <router-link-stub
          to="/archives/1"
        >
          Archive1 (6 texts |
      2 dossiers)
        </router-link-stub>
      </div>
      <div>
        <router-link-stub
          to="/archives/2"
        >
          Archive2 (3 texts |
      1 dossiers)
        </router-link-stub>
      </div>
      <div>
        <router-link-stub
          to="/archives/3"
        >
          Archive3 (30 texts |
      7 dossiers)
        </router-link-stub>
      </div>
      <div>
        <router-link-stub
          to="/archives/4"
        >
          Archive4 (20 texts |
      4 dossiers)
        </router-link-stub>
      </div>
      <div>
        <router-link-stub
          to="/archives/5"
        >
          Archive5 (15 texts |
      8 dossiers)
        </router-link-stub>
      </div>
      <div>
        <router-link-stub
          to="/archives/6"
        >
          Archive6 (21 texts |
      9 dossiers)
        </router-link-stub>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "emitted": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "html": [Function],
  "isUnmounted": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "unmount": [Function],
  "updateProps": [Function],
}
`;
